crate/cli/src/config.rs

```diff
@@ -53,6 +56,55 @@ fn not(b: &bool) -> bool {
     !*b
 }
 
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Clone)]
+pub struct TeeConf {
+    pub(crate) mr_enclave: Option<String>,
+    pub(crate) signing_key: Option<String>,
+    pub(crate) sev_measurement: Option<String>,
+}
+
+impl TryInto<TeeMeasurement> for TeeConf {
+    type Error = CliError;
+
+    fn try_into(self) -> Result<TeeMeasurement, Self::Error> {
+        match (
+            self.mr_enclave,
+            self.signing_key,
+            self.sev_measurement,
+            guess_tee(),
+        ) {
+            (Some(e), Some(s), None, Ok(TeeType::Sgx)) => Ok(TeeMeasurement::Sgx {
+                mr_signer: compute_mr_signer(&fs::read_to_string(s)?)
+                    .map_err(|e| CliError::Default(format!("MR signer computation failed: {e:?}")))?
+                    .as_slice()
+                    .try_into()
+                    .map_err(|_| CliError::Default("Bad MR signer bytes-size".to_string()))?,
```

Comment:
should we keep the error details?
```suggestion
                    .map_err(|e| CliError::Default(format!("Bad MR signer bytes-size: error: {e:?}"))?,
```

if we don't, maybe add a From conversion in the CLIError struct.

Reply:
Do you prefer a From conversion ? I always hesitate to do that for conversion error because I could be use elsewhere (transparently) and we need more contexts most of the time. That's why I use a `map_err` on such error. What do you think about it ?

Reply:
Probably a From conversion is enough here.